source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRRenderer.mm

```diff
@@ -0,0 +1,130 @@
+#import "ACRColumnView.h"
+
+
+using namespace AdaptiveCards;
+
+@implementation ACRRenderer
+
++ (UIView *)renderWithAdaptiveCards:(std::shared_ptr<AdaptiveCard> const &)adaptiveCard
+                             inputs:(NSMutableArray *)inputs
+                     viewController:(UIViewController *)vc
+                         guideFrame:(CGRect)guideFrame
+                         hostconfig:(std::shared_ptr<HostConfig> const &)config
+{
+    std::vector<std::shared_ptr<BaseCardElement>> body = adaptiveCard->GetBody();
+    
+    ACRColumnView *verticalView = nil;
+    
+    if(!body.empty())
+    {
+         verticalView = [[ACRColumnView alloc] initWithFrame:CGRectMake(0, 0, guideFrame.size.width, guideFrame.size.height)];
+        
+        [ACRRenderer render:verticalView inputs:inputs withCardElems:body andHostConfig:config];
+
+        std::vector<std::shared_ptr<BaseActionElement>> actions = adaptiveCard->GetActions();
+        UIView<ACRIContentHoldingView> *actionChildView = [ACRRenderer renderButton:vc inputs:inputs superview:verticalView actionElems:actions hostConfig:config];
+        [verticalView addArrangedSubview:actionChildView];
+    }
+    return verticalView;
+}
+
++ (UIView *)renderWithJson:(NSString *)str
+                    inputs:(NSArray *)inputs
+            viewController:(UIViewController *)vc
+                guideFrame:(CGRect)guideFrame
+                hostconfig:(std::shared_ptr<HostConfig> const &)config
+{
+    std::shared_ptr<AdaptiveCard> adaptiveCard = AdaptiveCard::DeserializeFromString(std::string([str UTF8String]));
+    return [ACRRenderer renderWithAdaptiveCards:adaptiveCard
+                                         inputs:(NSMutableArray *)inputs
+                                 viewController:vc
+                                     guideFrame:guideFrame
+                                     hostconfig:config];
+}
+
++ (UIView<ACRIContentHoldingView> *)renderButton:(UIViewController *)vc
+                                          inputs:(NSMutableArray *)inputs
+                                       superview:(UIView<ACRIContentHoldingView> *)superview
+                                     actionElems:(std::vector<std::shared_ptr<BaseActionElement>> const &)elems
+                                      hostConfig:(std::shared_ptr<HostConfig> const &)config
+{
+    ACRRegistration *reg = [ACRRegistration getInstance];
+    UIView<ACRIContentHoldingView> *childview = nil;
+    UILayoutConstraintAxis axis = UILayoutConstraintAxisVertical;
+    if(ActionsOrientation::Horizontal == config->actions.actionsOrientation)
+    {
+        childview = [[ACRColumnSetView alloc] initWithFrame:CGRectMake(0, 0, superview.frame.size.width, superview.frame.size.height)];
+        axis = UILayoutConstraintAxisHorizontal;
+    }
+    else
+    {
+        childview = [[ACRColumnView alloc] initWithFrame:CGRectMake(0, 0, superview.frame.size.width, superview.frame.size.height)];
+    }
+    
+    for(auto elem:elems)
```

Comment:
thanks!

---
In reply to: [132049954](https://github.com/Microsoft/AdaptiveCards/pull/532#discussion_r132049954) [](ancestors = 132049954)